spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -196,12 +196,6 @@ public final void setMissingQueuesFatal(boolean missingQueuesFatal) {
 	}
 
 	@Override
-	public void setQueues(Queue... queues) {
-		Assert.state(!isRunning(), "Cannot set queue names while running, use add/remove");
```

Comment:
Currently both `setQueues(Queue...)` and `setQueueNames(String...)` are overridden with the same checks.

However methods in the base class collect names from `Queue...` and call the corresponding method with `String...` - I tried to mentioned that above, sorry it it was not clear.
So, it's enough to have this extra logic only in the methods with `String...`.